### PR TITLE
Revert health to run in a single thread

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

This PR reverts health to run in a single thread for all hosts. It still retains the health initialization in the health thread as so to not delay host creation.

##### Test Plan

Will be tested that it performs as it should in regards also to the cloud.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

File: health/health.c [hunk 1]
```diff
@@ -1017,518 +1017,531 @@ void *health_main(void *ptr) {
```
Comment (stelfrag) [inline health/health.c:1017]: You don't need to get `now_realtime_sec()` here, it is done in the loop anyway

File: database/rrd.h [hunk 2]
```diff
@@ -1006,8 +1006,6 @@ struct rrdhost {
     // health monitoring options
 
     unsigned int health_enabled;                   // 1 when this host has health enabled
-    bool health_spawn;                             // true when health thread is running
-    unsigned int aclk_alert_reloaded;              // 1 on thread start and health reload, 0 after removed are sent
     time_t health_delay_up_to;                     // a timestamp to delay alarms processing up to
```
Comment (stelfrag) [inline database/rrd.h:1009]: Can group health related variables eg. in a 

```
 struct { 
    ...

 } health;
```
structure

File: health/health.c [hunk 3]
```diff
@@ -1017,518 +1017,531 @@ void *health_main(void *ptr) {
...
+            if (!health_running_logged) {
+                log_health("[%s]: Health is running.", rrdhost_hostname(host));
+                health_running_logged = true;
             }
 
-            runnable++;
-            rc->old_value = rc->value;
-            rc->run_flags |= RRDCALC_FLAG_RUNNABLE;
+            if(likely(!host->health_log_fp) && (loop == 1 || loop % cleanup_sql_every_loop == 0))
+                sql_health_alarm_log_cleanup(host);
```
Comment (stelfrag) [inline health/health.c:1110]: Lets have `sql_health_alarm_log_cleanup` use metadata thread to do the cleanup

File: health/health.c [hunk 4]
```diff
@@ -1017,518 +1017,531 @@ void *health_main(void *ptr) {
...
+
+                       if (unlikely(!(rc->rrdcalc_flags & RRDCALC_FLAG_DB_STALE))) {
+                       rc->rrdcalc_flags |= RRDCALC_FLAG_DB_STALE;
+                       error("Health on host '%s', alarm '%s.%s': database is stale", host->hostname, rc->chart?rc->chart:"NOCHART", rc->name);
+                       }
+                       }
+                       else if (unlikely(rc->rrdcalc_flags & RRDCALC_FLAG_DB_STALE))
+                       rc->rrdcalc_flags &= ~RRDCALC_FLAG_DB_STALE;
+                    */
+
```
Comment (stelfrag) [inline health/health.c:1224]: Lets delete the commented code

File: database/sqlite/sqlite_metadata.c [hunk 5]
```diff
@@ -685,6 +685,16 @@ static void check_dimension_metadata(struct metadata_wc *wc)
         error_report("Failed to finalize the prepared statement when reading dimensions");
 }
 
+static void cleanup_health_log(void)
+{
+    RRDHOST *host;
+    dfe_start_reentrant(rrdhost_root_index, host) {
+        if (rrdhost_flag_check(host, RRDHOST_FLAG_ARCHIVED) || !rrdhost_flag_check(host, RRDHOST_FLAG_METADATA_UPDATE))
```
Comment (stelfrag) [inline database/sqlite/sqlite_metadata.c:692]: Why check for RRDHOST_FLAG_METADATA_UPDATE ? This will probably skip cleanup

Comment (MrZammler) [reply database/sqlite/sqlite_metadata.c]: You are right, removed.

**Review by stelfrag — APPROVED** [review]

**Comment by MrZammler** [comment]
Thank you!

**Pull request merged**